.github/workflows/sycl_mlir_weekly.yml

```diff
@@ -0,0 +1,89 @@
+name: SYCL MLIR Weekly
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 22 * * THU'
+
+jobs:
+  ubuntu2204_build:
+    if: github.repository == 'intel/llvm'
+    uses: ./.github/workflows/sycl_linux_build.yml
+    secrets: inherit
+    with:
+      build_ref: sycl-mlir
+      build_cache_root: "/__w/"
+      build_artifact_suffix: default
+      merge_ref: ''
+      retention-days: 10
+
+  ubuntu2204_test:
+    needs: [ubuntu2204_build]
+    if: ${{ always() && !cancelled() && needs.ubuntu2204_build.outputs.build_conclusion == 'success' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Intel L0 GPU
+            runner: '["Linux", "gen12"]'
+            image: ghcr.io/intel/llvm/ubuntu2204_intel_drivers:latest
+            image_options: -u 1001 --device=/dev/dri --privileged --cap-add SYS_ADMIN
+            target_devices: ext_oneapi_level_zero:gpu
+            reset_gpu: true
+
+          - name: Intel OCL GPU
+            runner: '["Linux", "gen12"]'
+            image: ghcr.io/intel/llvm/ubuntu2204_intel_drivers:latest
+            image_options: -u 1001 --device=/dev/dri --privileged --cap-add SYS_ADMIN
+            target_devices: opencl:gpu
+            reset_gpu: true
+
+          - name: OCL CPU
+            runner: '["Linux", "x86-cpu"]'
+            image: ghcr.io/intel/llvm/ubuntu2204_intel_drivers:latest
+            image_options: -u 1001
+            target_devices: opencl:cpu
+
+    uses: ./.github/workflows/sycl_linux_run_tests.yml
+    with:
+      name: ${{ matrix.name }}
+      runner: ${{ matrix.runner }}
+      image: ${{ matrix.image }}
+      image_options: ${{ matrix.image_options }}
+      target_devices: ${{ matrix.target_devices }}
+      reset_gpu: ${{ matrix.reset_gpu }}
+      ref: ${{ github.sha }}
```

Comment:
I think this should be `sycl-mlir`. Or, more precisely, the commit sha of it at the time the build was being done. I think we can modify the build task to provide that in its output.

Reply:
I think we can put `sycl-mlir` here as per the [Checkout](https://github.com/marketplace/actions/checkout#usage) action docs

Reply:
Consider this
```
Time
|      * Build starts
|      * New commit in sycl-mlir branch, modifies both implementation and its e2e tests
|      * E2E tests start in a separate job, checking out the e2e tests
V
```

Now we have the implementation (from the build step) and the test out of sync.